__main__.py

```diff
@@ -39,15 +27,13 @@ def main():
     load_balancer = LoadBalancer(env, server, loadbalancer_conf, log_path)
     clients = Clients(env, load_balancer, client_conf, requests_conf, log_path)
 
-    env.process(flag(env, 5))
     SIM_DURATION_SECONDS = 12
     env.run(until=SIM_DURATION_SECONDS)
 
     logger = get_logger(log_path + "/main.log", "Main")
     log_server_data(logger, server)
-    log_latence(logger, clients.requests)
-
-    generate_results("results.log", "RESULTS", clients.requests)
+    log_latency(logger, clients.requests)
+    log_percentiles(logger, clients.requests, percentile)
```

Comment:
Why do you need to pass a pointer to a function? Wouldn't be much easier to just import the function that calculates the percentile? Or even better, the function percentile is only used in one place, thus, just move it there :)

Reply:
Thanks for comment, I have already changed that logic. Sorry about it.